clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -979,6 +978,10 @@ void EmitAssemblyHelper::EmitAssembly(BackendAction Action,
   if (LangOpts.SYCLIsDevice && LangOpts.SYCLExplicitSIMD)
     PerModulePasses.add(createGenXSPIRVWriterAdaptorPass());
 
+  if (llvm::Triple(TheModule->getTargetTriple()).isSPIR() &&
```

Comment:
I guess there should be a check/error reporting somewhere when this option is used with incompatible target.

Reply:
Done, thanks!